### ISSUE TITLE
Upgrade to 0.10.9 fails with "No Provider for MainContainerWillyWonka"

### ISSUE THREAD

**Issue by johnstill** [vmware/clarity#1609]
```
[ x] bug
[ ] feature request
[ ] enhancement
```

### Expected behavior
Upgrading should not complain about Willy Wonka.

### Actual behavior
My project builds correctly, but upon navigation to localhost:4200 the page hangs and I see the exception in the console.

### Reproduction of behavior
I've been able to reliably reproduce this on my machine, but unfortunately our project is pretty complex and there's no clear pointers in the traceback to anything in our code (and we're not messing with anything Willy Wonka, not directly at least). If I can get _some_ direction as to what might be causing this I'm willing to attempt generating minimal steps to reproduce.  At the moment I only know the error occurs with 0.10.9 but not 0.10.8 or below.

### Environment details

* **Angular version:** 4.x.x

* **Clarity version:** 0.10.9

* **OS and version:** OS X Sierra

* **Browser:** [N/A]

**Comment by mathisscott** [comment]
@youdz ^

**Comment by youdz** [comment]
This issue was closed automatically because we merged a pull request that addresses it, but I want to make clear what we decided to do.

Your issue comes from the fact that you're using a `clr-header` without a parent `clr-main-container`, which we do not support. Our entire layout needs to be written with a `clr-main-container` at the root because components like header or responsive nav depend on it.
So the PR just made the error message much more explicit by taking ownership of it, rather than let Angular throw without catching it.
